diff --git a/thirdparty/mDNSResponder-765.50.9/mDNSCore/anonymous.c b/thirdparty/mDNSResponder-765.50.9/mDNSCore/anonymous.c
index d2b80e7a..da87b1cb 100644
--- a/Build/mDNS/anonymous.c
+++ b/Build/mDNS/anonymous.c
@@ -20,6 +20,8 @@
 #include "anonymous.h"
 #include "DNSCommon.h"
 
+#define ANONYMOUS_DISABLED
+
 // Define ANONYMOUS_DISABLED to remove all the anonymous functionality
 // and use the stub functions implemented later in this file.
 
