diff --git a/thirdparty/mDNSResponder-765.50.9/mDNSCore/nsec3.c b/thirdparty/mDNSResponder-765.50.9/mDNSCore/nsec3.c
index 4e9e8c82..1d15c605 100644
--- a/Build/mDNS/nsec3.c
+++ b/Build/mDNS/nsec3.c
@@ -29,6 +29,8 @@
 // Define DNSSEC_DISABLED to remove all the DNSSEC functionality
 // and use the stub functions implemented later in this file.
 
+#define DNSSEC_DISABLED
+
 #ifndef DNSSEC_DISABLED
 
 typedef enum 
