diff --git a/thirdparty/mDNSResponder-765.50.9/mDNSCore/nsec.c b/thirdparty/mDNSResponder-765.50.9/mDNSCore/nsec.c
index 09d27356..516822ce 100644
--- a/Build/mDNS/nsec.c
+++ b/Build/mDNS/nsec.c
@@ -27,8 +27,9 @@
 
 // Define DNSSEC_DISABLED to remove all the DNSSEC functionality
 // and use the stub functions implemented later in this file.
+// Remove DNSSEC functionality for Windows builds
 
-#ifndef DNSSEC_DISABLED
+#ifndef DEFINE_WINDOWS_UNIVERSAL
 
 // Implementation Notes
 //
